### PR TITLE
[ACTION] PandaDoc: Get Document Status

### PR DESCRIPTION
**Pull request by zalmanlew**
### <samp>🤖 Generated by Copilot at 341a1d1</samp>

Added a new action to get document status and updated the `pandadoc.app.mjs` file with a new method to get document data. Incremented the versions of the `pandadoc` package and all the existing actions to reflect the changes and keep them consistent.

### <samp>🤖 Generated by Copilot at 341a1d1</samp>

*  Incremented the version of the `pandadoc` package to `0.2.2` to reflect the new features and updates in the actions and the app file ([link](https://github.com/PipedreamHQ/pipedream/pull/7675/files?diff=unified&w=0#diff-3d775c398b2ef6382327e8c78d5584393d3e8b93be1de20b410ed98299af2debL3-R3))
* Added a new action `get-document-status` to get basic status information about a document using its ID ([link](https://github.com/PipedreamHQ/pipedream/pull/7675/files?diff=unified&w=0#diff-4482355d94bb651da8cd879dd66e9c9ca0148d9836f877958432dcbb8f75cc9cR1-R22))
* Added a new method `getDocument` to the `pandadoc.app.mjs` file to make a request to the PandaDoc API to get detailed data about a document by its ID ([link](https://github.com/PipedreamHQ/pipedream/pull/7675/files?diff=unified&w=0#diff-74f4072e665d9faba2515b3e77fb8b8076788b4128dedbf4b47b77ca267e1d1dR191-R195))
* Incremented the versions of the existing actions that depend on the `pandadoc.app.mjs` file to keep them consistent with the updated app file ([link](https://github.com/PipedreamHQ/pipedream/pull/7675/files?diff=unified&w=0#diff-d2ae458abfce07e8071bcf0a30b93f89fb12586f0a5044c3c373dab7c8d0aa65L12-R12), [link](https://github.com/PipedreamHQ/pipedream/pull/7675/files?diff=unified&w=0#diff-81be0a37bd48e298303653a6a073359386b218490405b9f2a1ceb331681a03adL10-R10), [link](https://github.com/PipedreamHQ/pipedream/pull/7675/files?diff=unified&w=0#diff-e2858f07c8884fc18c7eb1d6712512c2d7463285e27bd3500b97690f13c851bcL8-R8), [link](https://github.com/PipedreamHQ/pipedream/pull/7675/files?diff=unified&w=0#diff-8b07fcf2c4b753bd913bb5efa0a62666d151a83ed68ec54f7c63f775280bfbe8L8-R8), [link](https://github.com/PipedreamHQ/pipedream/pull/7675/files?diff=unified&w=0#diff-be34986176481357fce7d6ff587d56df9f2f6ca552ba2f3f728b9ba50e30b989L8-R8), [link](https://github.com/PipedreamHQ/pipedream/pull/7675/files?diff=unified&w=0#diff-0dd6b87c01cfb40c754db5d3c05fa68fe82a69b419b3f754c3bc4373cf5e19c6L8-R8), [link](https://github.com/PipedreamHQ/pipedream/pull/7675/files?diff=unified&w=0#diff-8c632fdd9221e09d530dbd59dde0d16a34e7ed16f30c1563c378e7c2725902abL8-R8), [link](https://github.com/PipedreamHQ/pipedream/pull/7675/files?diff=unified&w=0#diff-5f9526a6fafe9220ce573537f678eae09f3112d33660a68ee34ee43e9d7e078aL8-R8), [link](https://github.com/PipedreamHQ/pipedream/pull/7675/files?diff=unified&w=0#diff-c4e5d66b3bd2a7bf68f29263126d1c90a7a66f3c480d43dc4696b6ca55d35ed1L9-R9), [link](https://github.com/PipedreamHQ/pipedream/pull/7675/files?diff=unified&w=0#diff-bdc3d8bc2358aea7dde5df409221916093d3aeeb72c1b344ef21c876271a83c7L8-R8), [link](https://github.com/PipedreamHQ/pipedream/pull/7675/files?diff=unified&w=0#diff-6e8f4cbc7cacd75282446d5370f4d60b1c7d5f0e0e3ffef7064c107f5ff93cc3L8-R8))

### CODE REVIEW

**Comment by pipedream-component-development** [comment]
Thank you so much for submitting this! We've added it to our backlog to review, and our team has been notified.

**Comment by pipedream-component-development** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:
- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible

File: components/pandadoc/actions/get-document-status/get-document-status.mjs [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import app from "../../pandadoc.app.mjs";
+
+export default {
+  key: "pandadoc-get-document-status",
+  name: "Get Document Status",
+  description:
+    "Get basic status info about a document. [See the docs here](https://developers.pandadoc.com/reference/document-status)",
```
Comment (lcaresia) [inline components/pandadoc/actions/get-document-status/get-document-status.mjs:7]:
```suggestion
    "Get basic status info about a document. [See documentation here](https://developers.pandadoc.com/reference/document-status)",
```

File: components/pandadoc/actions/get-document-status/get-document-status.mjs [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  props: {
+    app,
+    id: {
+      propDefinition: [app, "documentId"],
+    },
+  },
+  async run({ $ }) {
+    const response = await this.app.getDocument(this.id);
+
+    $.export("$summary", `Successfully fetched document statu with ID: ${id}`);
```
Comment (lcaresia) [inline components/pandadoc/actions/get-document-status/get-document-status.mjs:19]:
```suggestion
    $.export("$summary", `Successfully fetched document status with ID ${id}`);
```

File: components/pandadoc/actions/get-document-status/get-document-status.mjs [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  type: "action",
+  version: "0.0.1",
+  props: {
+    app,
+    id: {
+      propDefinition: [app, "documentId"],
+    },
+  },
+  async run({ $ }) {
+    const response = await this.app.getDocument(this.id);
```
Comment (lcaresia) [inline components/pandadoc/actions/get-document-status/get-document-status.mjs:17]:
```suggestion
    const response = await this.app.getDocument({
      id: this.id
    });
```

**Review by lcaresia — CHANGES_REQUESTED** [review]
Thanks for your contribution!

Just need some fixes

**Comment by zalmanlew** [comment]
@lcaresia changes done

**Review by lcaresia — DISMISSED** [review]
LGTM!

**Comment by zalmanlew** [comment]
@vunguyenhung something failed in the commit

**Comment by zalmanlew** [comment]
@vunguyenhung @lcaresia what's the status with this PR? Can it be merged?

**Comment by lcaresia** [comment]
@zalmanlew Hey, merged now, thanks!

**Pull request merged**